modules/candidate_parameters/ajax/getData.php

```diff
@@ -0,0 +1,353 @@
+}
+
+function getParticipantStatusFields() {
+    include_once __DIR__
+        . "/../../candidate_parameters/php/"
+        . "NDB_Form_candidate_parameters.class.inc";
+
+    $candID = $_GET['candID'];
+
+    $db =& Database::singleton();
+
+    // get pscid
+    $pscid = $db->pselectOne(
+        'SELECT PSCID FROM candidate where CandID = :candid',
+        array('candid' => $candID)
+    );
+
+    $statusOptions = NDB_Form_candidate_parameters::getParticipantStatusOptions();
+    $reasonOptions = array();
+
+    $required = $db->pselect('SELECT ID from participant_status_options where Required=1', array());
+    $parentIDs = $db->pselect('SELECT distinct(parentID) from participant_status_options', array());
+    $parentIDMap = array();
+
+    foreach ($parentIDs as $ID) {
+        $reasonOptions = array();
+        foreach ($ID as $parentID) {
+            if ($parentID != null) {
+                $options = $db->pselect(
+                    "SELECT ID, Description FROM participant_status_options WHERE parentID=:pid",
+                    array('pid' => $parentID)
+                );
+                foreach ($options as $option) {
+                    $reasonOptions[$option['ID']] = $option['Description'];
+                }
+                $parentIDMap[$parentID] = $reasonOptions;
+            }
+        }
+    }
+
+    $status = $db->pselectOne("SELECT participant_status FROM participant_status WHERE CandID=:candid", array('candid' => $candID));
+    $suboption = $db->pselectOne("SELECT participant_suboptions FROM participant_status WHERE CandID=:candid", array('candid' => $candID));
+    $reason = $db->pselectOne("SELECT reason_specify FROM participant_status WHERE CandID=:candid", array('candid' => $candID));
+
+    $history = getParticipantStatusHistory($candID);
+
+    $result = [
+        'pscid' => $pscid,
+        'candID' => $candID,
+        'statusOptions' => $statusOptions,
+        'required' => $required,
+        'reasonOptions' => $reasonOptions,
+        'parentIDs' => $parentIDMap,
+        'participant_status' => $status,
+        'participant_suboptions' => $suboption,
+        'reason_specify' => $reason,
+        'history' => $history
+    ];
+
+    return $result;
+}
+
+function getParticipantStatusHistory($candID) {
+    $db =& Database::singleton();
+    $unformattedComments = $db->pselect(
+        "SELECT (SELECT Description FROM participant_status_options pso WHERE ID=psh.participant_status) AS status, (SELECT Description from participant_status_options pso WHERE ID=psh.participant_subOptions) AS suboption, entry_staff, data_entry_date, reason_specify FROM participant_status_history psh WHERE CandID=:cid",
+        array('cid' => $candID)
+    );
+
+    $commentHistory = '';
+    foreach ($unformattedComments as $comment) {
+        $commentString = '';
+        $commentString .= '[' . $comment['data_entry_date'] . '] ';
+        $commentString .= '<i>Updated by ' . $comment['entry_staff'] . '. </i>';
```

Comment:
The HTML (and static text) should be somewhere in react/template not in PHP. (This is especially disliked by @driusan 💀 )